### PR TITLE
Extract tests by queries

### PR DESCRIPTION
**Pull request by MattAgn**
### Summary

This PR is the follow up of this [PR](https://github.com/callstack/react-native-testing-library/pull/608) and this [PR](https://github.com/callstack/react-native-testing-library/pull/654) addressing this [issue](https://github.com/callstack/react-native-testing-library/issues/325). 
Its purpose is to re-organize the tests of the queries byDisplayValue, byTestId, byText and byPlaceholderText and have them each in a separate file (for now they all are in the [findByApi.test.js](https://github.com/callstack/react-native-testing-library/blob/master/src/__tests__/findByApi.test.js) and [render.test.js](https://github.com/callstack/react-native-testing-library/blob/master/src/__tests__/render.test.js) files) .

### Test plan

Tests should not change and only be moved to the proper query files

### CODE REVIEW

**Comment by MattAgn** [comment]
@thymikee are those tests ok for you? is it the kind of refactor you had in mind?

**Comment by thymikee** [comment]
Yup, but since this touches tests, I need to take a double care in verifying we don't miss any use cases we covered. Struggling to find time at the moment, but will get there :)

**Comment by MattAgn** [comment]
Sure, totally understand! Tell me if I can be of any help :)

File: src/__tests__/byDisplayValue.test.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      defaultValue={DEFAULT_INPUT_CHEF}
+    />
+    <TextInput defaultValue={DEFAULT_INPUT_CUSTOMER} />
+    <TextInput defaultValue={'hello'} value="" />
+  </View>
+);
+
+test('getByDisplayValue, queryByDisplayValue', () => {
+  const { getByDisplayValue, queryByDisplayValue } = render(<Banana />);
+  const input = getByDisplayValue(/custom/i);
```
Comment (dsznajder) [inline src/__tests__/byDisplayValue.test.js:34]: 👀

**Review by thymikee — APPROVED** [review]
Thanks @dsznajder for the review

File: src/__tests__/byDisplayValue.test.js (hunk 1 above)
Comment (thymikee) [reply src/__tests__/byDisplayValue.test.js:34]: Feel free to address that as a followup

**Pull request merged**